be/src/block_cache/fb_cachelib.cpp

```diff
@@ -29,8 +30,19 @@ Status FbCacheLib::init(const CacheOptions& options) {
         config.enableNvmCache(nvmConfig);
     }
 
-    _cache = std::make_unique<Cache>(config);
-    _default_pool = _cache->addPool("default pool", _cache->getCacheMemoryStats().cacheSize);
+    try {
+        _cache = std::make_unique<Cache>(Cache::SharedMemAttach, config);
+        LOG(INFO) << "block cache is restored successfully";
+    } catch (const std::exception& e) {
+        // Attaching failed. Create a new one but make sure that
+        // the old cache is destroyed before creating a new one.
+        // This allows us to release any held resources (such as
+        // open file descriptors and associated fcntl locks).
+        _cache.reset();
```

Comment:
should we clean the directory?

Reply:
> should we clean the directory?

It is not necessary because the cache meta is reset and the disk file will be reused. But from user views, maybe it is useful to check the real data that has been cached on disks. So, Iet me add the procedure.